go.mod

```diff
@@ -1,10 +1,10 @@
 module github.com/containerd/nri
 
-go 1.22.0
+go 1.24.1
```

Comment:
We need to be a bit careful with this and check if this is (already) fine (at this point) for both for `containerd/main` and `containerd/release/2.0`. Ideally also for 1.7.x, but I'd say updating that anyway might require rolling a maintenance branch with hand-picked cherry-picks and we probably don't need/want to push WASM support for 1.7.x anyway.

Reply:
but this is forcing everyone importing this library to move to 1.24, no?

Reply:
> but this is forcing everyone importing this library to move to 1.24, no?

Yes.

We could think about wrapping a `// +build go1.24` tag, but this would make a feature dependent on a certain golang version. It would also require more boilerplate.

Reply:
nod, and the MUST be go 1.24.1 in the go.mod file is forcing everyone that includes this to be dependent on 1.24.1 or higher. This change by golang to make go the go dependency a MUST is crashing all backwards compatibility testing across the board. Typically you chase it down and someone thought it cool to use a new log feature just introduced.

Reply:
I see, I was more asking about if was a deliberate change or an oversight, it looks is the only way, I didn;t mean to push in one or other direction

Reply:
Per the community call today, we are not going to upgrade containerd 2.1 to Go 1.24.  This leaves us with either option (2) or (3) above.

My preference as-is would be option 3, disabling WASM support entirely until we have moved to Go 1.24 (which will happen for containerd 2.2).  Unless there is some way to have the new WASM support sans-tinygo support old tinygo-based plugins?